packages/apollo-language-server/src/providers/federation-info/endpoint.ts

```diff
@@ -0,0 +1,84 @@
+// IntrospectionSchemaProvider (http => IntrospectionResult => schema)
+// import { NotificationHandler } from "vscode-languageserver";
+import { execute as linkExecute, toPromise } from "apollo-link";
+import { createHttpLink, HttpLink } from "apollo-link-http";
+import {
+  GraphQLSchema,
+  buildClientSchema,
+  getIntrospectionQuery,
+  ExecutionResult,
+  IntrospectionQuery,
+  parse
+} from "graphql";
+import { Agent } from "http";
+import { fetch } from "apollo-env";
+import { RemoteServiceConfig } from "../../config";
+import {
+  // GraphQLSchemaProvider,
+  // SchemaChangeUnsubscribeHandler
+  GraphQLFederationInfoProvider,
+  FederationInfo
+} from "./base";
+
+export class EndpointFederationInfoProvider
+  implements GraphQLFederationInfoProvider {
+  private info?: FederationInfo;
+
+  constructor(private config: Exclude<RemoteServiceConfig, "name">) {}
+
+  private async getFederationInfo() {
+    const { skipSSLValidation, url, headers } = this.config;
+    const options: HttpLink.Options = {
+      uri: url,
+      fetch,
+      ...(skipSSLValidation && { fetchOptions: { agent: new Agent() } })
+    };
+
+    const getFederationInfoQuery = `
+      query getFederationInfo {

```

Comment:
Iiiiiinteresting...just thinking out loud. Seems we have a meta graphql problem here. Our type generation and VSCode are both going to try to validate this query against our schema 😂 

Might need an `// apollo-ignore` at some point.

Reply:
Not in this case :) There's no `gql` so it's happily ignored!